fairtools/MCConfigurator/FairXmlVMCConfig.cxx

```diff
@@ -0,0 +1,429 @@
+{
+    fMCEngine = mcEngine;
+    if (!((strcmp(mcEngine, "TGeant4") == 0) || (strcmp(mcEngine, "TGeant3") == 0))) {
+        LOG(fatal) << "FairXmlVMCConfig::Setup() Engine \"" << mcEngine << "\" unknown!";
+    }
+
+    string xmlFileName = ObtainXmlFileName(mcEngine);
+
+    // Parse xml file
+    // Only file with restricted xml syntax are supported
+    XMLDocPointer_t xmldoc = fXmlEngine->ParseFile(xmlFileName.data());
+    if (!xmldoc)
+        return;
+
+    // take access to main node
+    fXmlConfig = fXmlEngine->DocGetRootElement(xmldoc);
+
+    if (strcmp(mcEngine, "TGeant4") == 0) {
+        SetupGeant4();
+    } else if (strcmp(mcEngine, "TGeant3") == 0) {
+        SetupGeant3();
+    }
+    SetupStack();
+    SetCuts();
+
+    StoreXmlInfo();
+}
+
+void FairXmlVMCConfig::SetupPostInit(const char* mcEngine)
+{
+    if (!fPostInitFlag) {
+        LOG(info) << "FairXmlVMCConfig::SetupPostInit() OFF.";
+        return;
+    }
+
+    if (!((strcmp(mcEngine, "TGeant4") == 0))) {
+        LOG(fatal) << "FairXmlVMCConfig::SetupPostInit() only valid for TGeant4.";
+    }
+    LOG(info) << "Loading Geant4 PostInit Config.";
+
+    TString work = getenv("VMCWORKDIR");
+    TString work_config = work + "/gconfig/";
+    work_config.ReplaceAll("//", "/");
+
+    TString config_dir = getenv("CONFIG_DIR");
+    config_dir.ReplaceAll("//", "/");
+
+    Bool_t AbsPath = kFALSE;
+
+    TString ConfigMacro;
+    TString g4Macro;
+    if (fPostInitName.empty()) {
+        g4Macro = "g4ConfigPostInit.C";
+        fPostInitName = g4Macro;
+    } else {
+        if (fPostInitName.find("/") != std::string::npos) {
+            AbsPath = kTRUE;
+        }
+        g4Macro = fPostInitName;
+        LOG(debug) << "---------------User config is used: " << g4Macro.Data();
+    }
+    if (!AbsPath && TString(gSystem->FindFile(config_dir.Data(), g4Macro)) != TString("")) {
+        LOG(debug) << "---User path for Configuration (" << fPostInitName << ") is used: " << config_dir.Data();
+        ConfigMacro = g4Macro;
+    } else {
+        if (AbsPath) {
+            ConfigMacro = fPostInitName;
+        } else {
+            ConfigMacro = work_config + fPostInitName;
+        }
+    }
+
+    XMLDocPointer_t xmldoc = fXmlEngine->ParseFile(ConfigMacro.Data());
+    if (!xmldoc)
+        return;
+
+    // take access to main node
+    fXmlConfigPostInit = fXmlEngine->DocGetRootElement(xmldoc);
+
+    XMLNodePointer_t xmlNode = 0;
+
+    if ((xmlNode = GetPointer("SIM:G4_PostInit:Commands", fXmlConfigPostInit))) {
+        std::vector<string> strVect = ConvertToVector<string>(xmlNode);
+        LOG(debug) << "FairXmlVMCConfig: SIM:G4_PostInit:Commands got " << strVect.size() << " values:";
+        TGeant4* geant4 = dynamic_cast<TGeant4*>(TVirtualMC::GetMC());
+        for (auto val : strVect) {
+            LOG(debug) << "   -> \"" << val << "\"";
+            geant4->ProcessGeantCommand(val.data());
+        }
+    }
```

Comment:
* Please use descriptive names describing the variable content and **not** the type
* Make local variables `const` by default

```cpp
    if ((xmlNode = GetPointer("SIM:G4_PostInit:Commands", fXmlConfigPostInit))) {
        std::vector<string> const commands = ConvertToVector<string>(xmlNode);
        LOG(debug) << "FairXmlVMCConfig: SIM:G4_PostInit:Commands got " << commands.size() << " values:";
        TGeant4* geant4 = dynamic_cast<TGeant4*>(TVirtualMC::GetMC());
        for (auto const command : commands) {
            LOG(debug) << "   -> \"" << command << "\"";
            geant4->ProcessGeantCommand(command.data());
        }
    }
```